Automattic/jetpack
# Search: Apply URL param just_upgraded for tooltips after upgrading to a paid plan

#### Changes proposed in this Pull Request:

* Introduce URL param `just_upgraded` as the flag to trigger tooltips after upgrading to a paid plan.
* Introduce localStorage item `upgrade_tooltip_finished` to stop tooltips after users have dismissed them even though there is a `just_upgraded` flag.

#### Other information:


#### Jetpack product discussion


p1HpG7-hEH-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






> Note: Since the upgrade path from plan free to plan paid is not working presently, we must test it via the paid plan purchase flow. I believe they are the same thing in the redirected URL way.

* [Launch a new site with `Jetpack Search` without a Jetpack Search subscription](https://jurassic.ninja/create/?jetpack-beta&branches.jetpack-search=update/upgrade_tooltips_triggering).
* Navigate to the site with URL `/wp-admin/admin.php?page=jetpack-search&new_pricing_202208=1`.
* Press the `Get Search` button to purchase a `paid` Jetpack Search plan.
* Complete the checkout flow and ensure the URL of the redirected page contains `&just_upgraded=1`.
* Please manually append the URL param `&new_pricing_202208=1` and reload the page.
* Ensure there are upgraded usage limit tooltips displayed above the DonutMeter.
* Press the `Next` and coming `Finish` buttons on the tooltips.
* Ensure the tooltips are gone now.
* Reload the page without any URL changes.
* Ensure the tooltips are not showing anymore.
